server/controllers/userController.js

```diff
@@ -211,6 +213,36 @@ class UserController {
     );
     return util.successStatus(res, 200, 'Password reset successfully');
   }
+  static async assignUserRole(req, res) {
+    const { id } = req.user ? req.user : req.query;
```

Comment:
Where is this coming from?

Reply:
User id should never come from a query in this case, the token used for the request is a better option to prevent using other users details to make changes

Reply:
Ok, noted